app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,79 @@
+class ProjectsController < ApplicationController
+  before_action :set_project, only: [:show, :edit, :update, :destroy]
+  before_action :validates_role, only: [:new, :edit, :update, :create]
+  before_action :validates_admin, only: :destroy
+
+  def index
+    @projects = Project.all
+  end
+
+  def show
+  end
+
+  def new
+    @project = Project.new
+    get_clients
```

Comment:
these 2 can be moved to a method and then called here.
That method would also be used when save fails on update or create and you render stuff.